src/silo/preprocessing/metadata_info.cpp

```diff
@@ -9,137 +9,173 @@
+      throw silo::preprocessing::PreprocessingException(error_message);
+   }
+
+   std::set<std::string> actual_fields;
    for (size_t idx = 0; idx < result->ColumnCount(); idx++) {
-      file_metadata_fields[result->ColumnName(idx)] = "\"" + result->ColumnName(idx) + "\"";
+      actual_fields.emplace(result->ColumnName(idx));
+      if (std::find_if(database_config.schema.metadata.begin(), database_config.schema.metadata.end(), [&](const auto& metadata) {
+             return metadata.name == result->ColumnName(idx);
+          }) == database_config.schema.metadata.end()) {
+         SPDLOG_WARN(
+            "The field '{}' which is contained in the metadata file '{}' is not contained in the "
+            "database config.",
+            result->ColumnName(idx),
+            metadata_file.string()
+         );
+      }
    }
-   const std::unordered_map<std::string, std::string> validated_metadata_fields =
-      validateFieldsAgainstConfig(file_metadata_fields, database_config);
 
-   return {validated_metadata_fields};
+   for (const auto& field : database_config.schema.metadata) {
+      if (!actual_fields.contains(field.name)) {
+         const std::string error_message = fmt::format(
+            "The field '{}' which is contained in the database config is not contained in the "
+            "input field '{}'.",
+            field.name,
+            metadata_file.string()
+         );
+         SPDLOG_ERROR(error_message);
+         throw silo::preprocessing::PreprocessingException(error_message);
+      }
+   }
 }
 
-MetadataInfo MetadataInfo::validateFromNdjsonFile(
+void MetadataInfo::validateNdjsonFile(
    const std::filesystem::path& ndjson_file,
    const silo::config::DatabaseConfig& database_config
 ) {
    duckdb::DuckDB duck_db(nullptr);
    duckdb::Connection connection(duck_db);
 
    auto result = connection.Query(fmt::format(
-      "SELECT json_keys(metadata) "
-      "FROM read_json_auto(\"{}\") LIMIT 1; ",
+      "SELECT metadata.* "
+      "FROM read_json_auto(\"{}\") LIMIT 0; ",
       ndjson_file.string()
    ));
+
    if (result->HasError()) {
-      throw silo::preprocessing::PreprocessingException(
-         "Preprocessing exception when retrieving the field 'metadata', "
-         "duckdb threw with error: " +
+      SPDLOG_WARN(
+         "Preprocessing exception when retrieving the fields of the struct 'metadata' from the "
+         "metadata ndjson file '{}', "
+         "duckdb threw with error: {}",
+         ndjson_file.string(),
          result->GetError()
       );
-   }
-   if (result->RowCount() == 0) {
-      throw silo::preprocessing::PreprocessingException(fmt::format(
-         "File {} is empty, which must not be empty at this point", ndjson_file.string()
-      ));
-   }
-   if (result->RowCount() > 1) {
-      throw silo::preprocessing::PreprocessingException(
-         "Internal exception, expected Row Count=1, actual " + std::to_string(result->RowCount())
-      );
+      return;
```

Comment:
Is it ok to `return` here instead of throwing?

Reply:
True. Let's better have the emptiness check outside